src/mpi/coll/include/coll_impl.h

```diff
@@ -0,0 +1,150 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2006 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ *  Portions of this code were written by Intel Corporation.
+ *  Copyright (C) 2011-2017 Intel Corporation.  Intel provides this material
+ *  to Argonne National Laboratory subject to Software Grant and Corporate
+ *  Contributor License Agreement dated February 8, 2012.
+ */
+
+/*
+=== BEGIN_MPI_T_CVAR_INFO_BLOCK ===
+
+cvars:
+    - name        : MPIR_CVAR_PROGRESS_MAX_COLLS
+      category    : COLLECTIVE
+      type        : int
+      default     : 8
+      class       : device
+      verbosity   : MPI_T_VERBOSITY_USER_BASIC
+      scope       : MPI_T_SCOPE_ALL_EQ
+      description : >-
+        Maximum number of collective operations at a time that the progress engine should make progress on
+
+=== END_MPI_T_CVAR_INFO_BLOCK ===
+*/
+
+
+#ifndef MPIC_COLL_IMPL_H_INCLUDED
+#define MPIC_COLL_IMPL_H_INCLUDED
+
+#include "queue.h"
+
+extern MPIC_progress_global_t MPIC_progress_global;
+extern MPIC_global_t MPIC_global_instance;
+
+#define GLOBAL_NAME MPIC_
+
+#include "../transports/stub/transport.h"
+#include "../transports/mpich/transport.h"
+
+/* Algorithms with STUB transport */
+#include "../transports/stub/api_def.h"
+#include "../algorithms/stub/post.h"
+#include "../algorithms/tree/post.h"
+#include "../src/tsp_namespace_undef.h"
+
+
+/* Algorithms with MPICH transport */
+#include "../transports/mpich/api_def.h"
+#include "../algorithms/stub/post.h"
+#include "../algorithms/tree/post.h"
+#include "../src/tsp_namespace_undef.h"
+
+/*Default Algorithms*/
+#include "../algorithms/default/post.h"
```

Comment:
@pavanbalaji the default algorithms have been moved to .h files and included here as part of the unified collective framework. IMO, this conversion to static inline of the default algos is overkill, as they do not function as templates like `#include` lines above. We could simply include the decls here and leave the implementation in .c files. What do you think?

Reply:
I made these changes. Changed static inlines to regular functions for the default algorithms.